src/test/unit/controls/Line/LineLoad-spec.js

```diff
@@ -1226,6 +1226,91 @@ describe("Line - Load", () => {
+                    showElement: true,
+                }
+                graphDefault.loadContent(new Line(input));
+                const legendContainer = fetchElementByClass(
+                    lineGraphContainer,
+                    styles.legendItem
+                );
+                const legendItems = legendContainer.children;
+                expect(legendContainer).not.toBeNull();
+                expect(legendContainer.tagName).toBe("LI");
+                expect(legendItems.length).toBe(2);
+                const legendItem = document.body.querySelector(
+                    `.${styles.legendItem}`
+                );
+                expect(legendItem.getAttribute("aria-disabled")).toBe("true");
+                expect(legendItem.getAttribute("aria-current")).toBe("true");
+            })
+        });
+        describe("if legend has data and user sets showElement to be false", () => {
+            it("should show the legend", () => {
+                const input = getInput(valuesDefault, false, false);
+                input.legendOptions = {
+                    showElement: true,
+                }
+                graphDefault.loadContent(new Line(input));
+                const legendContainer = fetchElementByClass(
+                    lineGraphContainer,
+                    styles.legendItem
+                );
+                const legendItems = legendContainer.children;
+                expect(legendContainer).not.toBeNull();
+                expect(legendContainer.tagName).toBe("LI");
+                expect(legendItems.length).toBe(2);
+                const legendItem = document.body.querySelector(
+                    `.${styles.legendItem}`
+                );
+                expect(legendItem.getAttribute("aria-disabled")).toBe("false");
+                expect(legendItem.getAttribute("aria-current")).toBe("true");
+            })
+        });
+        describe("if legend has data and user sets showElement to be true", () => {
+            it("should show the legend", () => {
+                const input = getInput(valuesDefault, false, false);
+                input.legendOptions = {
+                    showElement: true,
+                }
+                graphDefault.loadContent(new Line(input));
+                const legendContainer = fetchElementByClass(
+                    lineGraphContainer,
+                    styles.legend
+                );
+                const legendItems = legendContainer.children;
+                expect(legendContainer).not.toBeNull();
+                expect(legendContainer.tagName).toBe("UL");
+                expect(legendItems.length).toBe(1);
+                const legendItem = document.body.querySelector(
+                    `.${styles.legendItem}`
+                );
+                expect(legendItem.getAttribute("aria-disabled")).toBe("false");
+                expect(legendItem.getAttribute("aria-current")).toBe("true");
+            });
+        });
```

Comment:
These tests can also be combined:

```js
describe("when the legend has data",  ()=>{
   it ("should hide the legend if showElement is false", ()=>{....})
        ......
   it ("should show the legend if showElement is true", ()=>{....})
        ......
}
```

Reply:
Updated [here](https://github.com/NarasimhaShenoi/carbon-graphs/commit/e5eae0edb49ef09342b04de3df0500d909b9e134)